### PR TITLE
ws_transport_options inserted at wrong level - WSS fails on non-443 TLS ports

### PR DESCRIPTION
**Pull request by h6w**
In order for wss to work, the contents of ws_transport_options needs to be at the same level as other ConnOpts.

This is required because gun requires two overrides for any non-standard TLS port:
```
{ ws_transport_options, [
                            {transport, tls},
                            {protocols,[http]}
                        ]}
```

The first option is required to specify tls (obviously) since gun will only enable tls on port 443 due to this code in gun.erl:
```
default_transport(443) -> tls;
default_transport(_) -> tcp.

```
Any non-443 SSL connection would then require the "{transport, tls}" option added to ws_transport_options.

The second option is required because gun defaults to http2, which then fails.   Reducing the protocols to just http allows tls to succeed and thus wss will succeed.

One or both of these may become unnecessary in future versions of gun (although I did test up to the current version and both are still issues), however being able to pass correct transport options to ConnOpts in the websocket connection will, no doubt, remain important regardless.

### CODE REVIEW

**Comment by qzhuyan** [comment]
pls rebase onto latest master.

**Comment by h6w** [comment]
I believe I have rebased this now.  Thanks.

**Comment by qzhuyan** [comment]
@h6w  thanks for the PR!

File: src/emqtt_ws.erl [hunk 1]
```diff
@@ -40,11 +40,11 @@ connect(Host0, Port, Opts, Timeout) ->
     {ok, _} = application:ensure_all_started(gun),
     %% 1. open connection
     TransportOptions = proplists:get_value(ws_transport_options, Opts, []),
-    ConnOpts = #{connect_timeout => Timeout,
+    TransportOpts = maps:from_list(TransportOptions),
```
Comment (zmstone) [inline src/emqtt_ws.erl:43]: it should be `TransportOpts = #{transport_opts => TransportOptions},` ?
I tried to fix it here, and verified in CI.
https://github.com/emqx/emqtt/pull/218/files#diff-8ec1346450794b68c5e1451510579efbbe75e28c4ef218329434a128402fcd09R43

**Comment by zmstone** [comment]
I accidently reverted this in one of the recent PRs. 
https://github.com/emqx/emqtt/pull/218/files#diff-8ec1346450794b68c5e1451510579efbbe75e28c4ef218329434a128402fcd09R43

will revert it back to this implementation and fix the test case.

**Pull request merged**